modules/exoplanets/datastructure.cpp

```diff
@@ -0,0 +1,332 @@
+#include <openspace/util/distanceconstants.h>
+#include <openspace/util/time.h>
+#include <openspace/util/timeconversion.h>
+#include <ghoul/logging/logmanager.h>
+#include <string_view>
+
+namespace {
+    constexpr std::string_view _loggerCat = "ExoplanetsModule";
+
+    // KeplerTranslation requires angles in range [0, 360]
+    float validAngle(float angle, float defaultValue, bool& usedDefault) {
+        if (std::isnan(angle)) {
+            usedDefault = true;
+            return defaultValue;
+        }
+        if (angle < 0.f) { return angle + 360.f; }
+        if (angle > 360.f) { return angle - 360.f; }
+        return angle;
+    };
+
+    // This is the documentation for the data object that is returned from the
+    // `openspace.exoplanets.systemData` function in the Scripting API.
+    //
+    // The data object provides the information needed to create the scene graph nodes
+    // for an exoplanet system. However, depending on what data is available for the
+    // specific planets and star, some values may be undefined.
+    struct [[codegen::Dictionary(ExoplanetSystemData)]] Parameters {
+        // The identifier of the system, which equals the identifier of the host star
+        // created based on the star name
+        std::string systemId;
+
+        // The position of the host star, expressed in meters
+        glm::dvec3 position;
+
+        // The number of known planets in the system that have enough data to be
+        // visualized
+        int numPlanets;
+
+        // The name of the star
+        std::string starName;
+
+        // The radius of the star, in meters, if data exists
+        std::optional<double> starRadius;
+
+        // An RGB color for the host star, computed from the star's B-V value, if data
+        // exists
+        std::optional<glm::dvec3> starColor [[codegen::color()]];
+
+        // The effective temperature of the star, in Kelvin, if data exists
+        std::optional<double> starTeff;
+
+        // The luminosity for the star, in units of solar luminosities [log10(Solar)], if
+        // data exists
+        std::optional<double> starLuminosity;
+
+        struct Planet {
+            // An identifier to use for the planet, created based on the planet name
+            std::string id;
+
+            // The name of the exoplanet
+            std::string name;
+
+            // The radius of the planet, in meters, if data exists
+            std::optional<double> starRadius;
+
+            // The semi-major axis of the planet's orbit, in meters. That is, the orbit
+            // size. This is required to be able to be visualize the planetary orbit
+            double semiMajorAxis;
+
+            // The uncertainty of the semi-major axis, given as an asymmetric uncertainty
+            // with two values: the lower and upper uncertainty for the orbit size. The
+            // values are given as values relative to the size of the SemiMajorAxis. For
```

Comment:
```suggestion
            // values are given as values relative to the size of the `SemiMajorAxis`. For
```

?

Reply:
Yes, I agree with this.  We should always use the code style for the member names, since this is how they show up in the documentation :)